QuadrupedalRobots/ETGRL/deployment/envs/EnvWrapper.py

```diff
@@ -0,0 +1,336 @@
+
+    def get_observation(self):
+        self.robot.ReceiveObservation()
+        sensors_dict = {}
+        if "motor" in self.sensor_mode:
+            motor_angle = self.robot.GetMotorAngles()
+            if self.normal:
+                motor_angle = (np.array(motor_angle)-np.array([0,0.9,-1.8]*4))/0.1
+            if self.sensor_mode["motor"] == 1:
+                motor_vel = self.robot.GetMotorVelocities()
+                if self.normal:
+                    motor_vel = (np.array(motor_vel)-np.zeros(12))/1.0
+                sensors_dict["MotorAngleAcc"] = np.concatenate((motor_angle,motor_vel),axis=0)
+            elif self.sensor_mode["motor"] == 2:
+                sensors_dict["MotorAngle"] = motor_angle
+
+        if "dis" in self.sensor_mode and self.sensor_mode["dis"]:
+            sensors_dict["BaseDisplacement"] = self.robot.GetBaseVelocity()
+
+        if "imu" in self.sensor_mode:
+            if self.first_reset:
+                self.first_rpy = np.array(self.robot.GetTrueBaseRollPitchYaw())
+                rpy = np.zeros(3)
+                self.first_reset = False
+            else:
+                rpy = np.array(self.robot.GetTrueBaseRollPitchYaw())-self.first_rpy
+            drpy = self.robot.GetTrueBaseRollPitchYawRate()
+            if self.normal:
+                rpy = (rpy-np.zeros(3))/0.1
+                drpy = (np.array(drpy)-np.zeros(3))/0.5
+            if self.sensor_mode["imu"]==1:
+                sensors_dict["IMU"] = np.concatenate((rpy,drpy),axis=0)
+            elif self.sensor_mode["imu"]==2:
+                sensors_dict["IMU"] = drpy
+
+        if "contact" in self.sensor_mode and self.sensor_mode["contact"]:
+            contact = self.robot.GetFootContacts()
+            sensors_dict["FootContactSensor"] = contact
+
+        observation_dict = collections.OrderedDict(sorted(list(sensors_dict.items())))
+        observations = []
+        for key, value in observation_dict.items():
+            observations.append(np.asarray(value).flatten())
+        flat_observations = np.concatenate(observations)
+        if "ETG" in self.sensor_mode and self. sensor_mode["ETG"]:
+            ETG_output = self.ETG_data[self.iter]
+            if self.normal:
+                ETG_output = (ETG_output-self.ETG_mean)/self.ETG_std
+            flat_observations = np.concatenate((flat_observations,ETG_output),axis=0)
+        observation_dict["real_action"] = self.last_action
+        return flat_observations,observation_dict
+
+    def reset(self,**kwargs):
+        self.first_reset = True
+        self.iter = 0
+        self.robot.Reset()     
+        obs,info = self.get_observation()
+        return obs,info
+    
+    def step(self,action,**kwargs):
+        self.robot.Step(action, robot_config.MotorControlMode.POSITION)
+        self.last_action = action
+        self.iter += 1
+        # obs,info = self.get_observation()
+        # info["real_action"] = action
+        # return obs,info
```

Comment:
Does the above code work as expected? It looks strange that the step function doesn't return the obs.

Reply:
yes, it worked as expected. It's modified and the obs is attained from self.getobs() to save time.